rust/src/arrow/linalg.rs

```diff
@@ -254,7 +251,7 @@ impl MatrixView {
         let mut builder = Float32Builder::with_capacity(n * dim);
         for idx in chosen.iter() {
             let s = self.data.slice(idx * dim, dim);
-            builder.append_slice(as_primitive_array::<Float32Type>(s.as_ref()).values());
+            builder.append_slice(s.values());
```

Comment:
Slice is strong typed now?

Reply:
yeah, must be part of the effort towards arrow-rs arrow2 merging?